config/config-api/src/com/thoughtworks/go/config/SecurityConfig.java

```diff
@@ -126,9 +126,12 @@ public boolean hasSecurityMethodChanged(SecurityConfig newSecurity) {
         if (newSecurity == null) {
             return true;
         }
+
         boolean ldapChanged = !ObjectUtil.equal(ldapConfig, newSecurity.ldapConfig());
         boolean passwordFileChanged = !ObjectUtil.equal(passwordFileConfig, newSecurity.passwordFileConfig());
-        return ldapChanged || passwordFileChanged;
+        final boolean securityAuthConfigChanged = !ObjectUtil.equal(securityAuthConfigs, newSecurity.securityAuthConfigs());
+
+        return ldapChanged || passwordFileChanged || securityAuthConfigChanged;
```

Comment:
I think any changes to security config should not affect oAuth, so wondering if we need to clean the tokens at if there is a change in security. But, we need to give a thought bout the expected behaviour when security is enabled and disabled.